File: oddcoin.py
def solve(queries):
    results = []
    for xi, ti in queries:
        denominations = [1, 3, 5, 10, 30, 50, 100, 300, 500, 1000, 3000, 5000, 10000, 30000, 50000]
        denominations.sort(reverse=True)
        count = float('inf')
        ticp = ti
        for i in range(0, 50000):
            if ti < i * xi:
                break
            print(i)
            ti = ticp
            count1 = i
            ti -= i * xi
            coinsans = []
            for coins in denominations:
                #while we can still subtract coin
                while ti >= coins:
                    coinsans.append(coins)
                    ti -= coins
                    count1 += 1
            print(*denominations)
            count = min(count, count1)
        results.append(count)
        print()
    return results

Q = int(input())
queries = []

for _ in range(Q):
    xi, ti = map(int, input().split())
    queries.append((xi, ti))

results = solve(queries)

for result in results:
    print(result)

File: oddcoin.cpp
#include<bits/stdc++.h>

using namespace std;
#define int long long
const int coins[] = {50000, 30000, 10000, 5000, 3000, 1000, 500, 300, 100, 50, 30, 10, 5, 3, 1};
signed solve() {    
    int a, b;
    cin >> a >> b;
    int ans = LONG_LONG_MAX;
    for(int i = 0; i <= 5e4; i++) {
        int t = b - a * i;
        if (t < 0) break;
        cout << t << endl;
        int count = i;
        for(int i = 0; i < 15; i++) {
            if(t >= coins[i]) cout << coins[i] << ' ';
            while(t >= coins[i]) {
                t -= coins[i];
                count++;
            }
        }
        cout << endl;
        ans = min(ans, count);
    }
    return ans;
}
signed main() {
    int q;
    cin >> q;
    for(int i = 1; i <= q; i++) {
        cout << solve() << '\n';
    }
}